src/main/core/controller.c

```diff
@@ -443,17 +507,26 @@ gint controller_run(Controller* controller) {
     _controller_initializeTimeWindows(controller);
 
     ConfigurationShadowElement* element = configuration_getShadowElement(controller->config);
-    g_assert(element && element->preloadPath.isSet);
+
+    gchar* preloadPath = _controller_scanRPathForPreloadShim();
+    if (preloadPath != NULL) {
+        message("found %s at %s", PRELOAD_SHIM_LIB_STR, preloadPath);
+    } else {
+        critical("could not find %s in rpath", PRELOAD_SHIM_LIB_STR);
+        return 1;
+    }
```

Comment:
Unless there is some advantage to doing this in the controller that I'm missing, I think the right place to do it is in the manager. Eventually, when we enhance shadow to utilize multiple machines, we'll have one manager per machine but only one single controller per simulation. Each machine (i.e., manager) will have to scan the rpath separately.

I think we can just move all of the code over to manager, and initiate the rpath scan in `manager_new`?

Reply:
Moved this over to the manager.